sql/core/src/test/scala/org/apache/spark/sql/TPCDSQueryTestSuite.scala

```diff
@@ -149,7 +187,12 @@ class TPCDSQueryTestSuite extends QueryTest with TPCDSBase with SQLQueryTestHelp
         classLoader = Thread.currentThread().getContextClassLoader)
       test(s"$name-v2.7") {
         val goldenFile = new File(s"$baseResourcePath/v2_7", s"$name.sql.out")
-        runQuery(queryString, goldenFile)
+        val (schema, output) = runQuery(queryString, goldenFile)
+        if (!regenerateGoldenFiles) {
+          shuffleConfSet.foreach { testConf =>
+            runQueryWithShuffleConf(queryString, schema, output, testConf)
+          }
+        }
```

Comment:
I prefer we do this:
1. change the `runQuery(query: String, goldenFile: File)` to `runQuery(query: String, goldenFile: File, conf: Map[String, String])`
2. if (regenerateGoldenFiles) { runQuery } else { runQuery * 4 } // default conf + 3 shuffle conf
3. if runQuery under default conf, no need to sort